### PR TITLE
Use some library functions

### PR DESCRIPTION
**Pull request by ajs6f**

### CODE REVIEW

**Comment by ajs6f** [comment]
Just replacing some homegrown stuff with library functions that came out since, and pruning two classes, one for which an equivalent is now available in library code, and the other wasn't used at all and never has been AFAICT.

**Comment by ajs6f** [comment]
Commits are orthogonal. I'll wait until the top of the week in case anyone has any comments/objections.

**Comment by ajs6f** [comment]
Also will send a further commit getting rid of calls to now-deprecated functions.

File: jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java [hunk 1]
```diff
@@ -33,32 +35,34 @@ private StrUtils() {}
...
+        return String.join("\n", args);
     }
 
-    /** Concatentate strings, using a separator */
+    /** 
+     * Concatentate strings, using a separator
+     * 
+     * @deprecated Prefer String.join(sep, args)
+     */
+    @Deprecated
```
Comment (afs) [inline jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java]: If we are going to deprecate this, then I really would like to see the changes propagated through the codebase, not left pending.

Currently, at least for me, Jena is warning free in the main modules so I can see when warnings appear. Having unfinished deprecations means that it is harder to notice warnings appearing.

File: jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java [hunk 2]
```diff
@@ -33,32 +35,34 @@ private StrUtils() {}
...
     public static String strjoinNL(List<String> args) {
-        return join("\n", args);
+        return String.join("\n", args);
     }
 
-    /** Concatentate strings, using a separator */
+    /** 
+     * Concatentate strings, using a separator
+     * 
+     * @deprecated Prefer String.join(sep, args)
```
Comment (afs) [inline jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java]: Add to javadoc `This function will be removed - do not use.`

File: jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java [hunk 3]
```diff
@@ -33,32 +35,34 @@ private StrUtils() {}
...
+        return String.join(sep, args);
     }
 
-    /** Concatentate strings, using a separator */
+    /**
+     * Concatentate strings, using a separator
+     * 
+     * @deprecated Prefer String.join(sep, args)
+     */
+    @Deprecated
```
Comment (afs) [inline jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java]: Ditto.

File: jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java [hunk 4]
```diff
@@ -85,17 +89,20 @@ public static int strCompareIgnoreCase(String s1, String s2) {
...
     }
     
+    /**
+     * @param x
+     * @return &lt;null&gt; if x == null, otherwise, x.toString()
+     * 
+     * @deprecated prefer {@link Objects#toString(Object, String)} with "&lt;null&gt;" or just {@link Objects#toString(Object)} 
+     */
+    @Deprecated
     public static String str(Object x) {
```
Comment (afs) [inline jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java]: I suggest we leave this, and not `@Deprecated`.  `str()` is supposed to be reminiscent of SPARQL `STR`.  `str(...)` is used elsewhere so this is a pattern.

**Comment by ajs6f** [comment]
@afs , yep there's another commit coming getting rid of those warnings. Just an oversight on my part, sorry.

File: jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java (hunk 4 above)
Comment (ajs6f) [inline jena-base/src/main/java/org/apache/jena/atlas/lib/StrUtils.java]: all right, if that seems clearer.

**Comment by ajs6f** [comment]
Weird, Travis blew out with No Heap Space...

https://api.travis-ci.org/jobs/186431648/log.txt?deansi=true

**Comment by ajs6f** [comment]
Yeah, it does a perfectly nice full build (no profile) for me. Not sure what Travis' problem is. I never even noticed that `src-examples` dir-- I'll add a commit to clean that up.

**Comment by ajs6f** [comment]
I'll wait for any further comment until Tuesday.

**Comment by ajs6f** [comment]
Thanks, I got caught up with holiday stuff and forgot to merge this.

**Pull request merged**